engineioxide/src/engine.rs

```diff
@@ -385,10 +442,28 @@ impl<H: EngineIoHandler> EngineIo<H>
             p => Err(Error::BadPacket(p))?,
         };
 
+        cfg_if! {
+            if #[cfg(feature = "v3")] {
+                // send a NOOP packet to any pending polling request so it closes gracefully
+                // V3 protocol introduce _paused_ polling transport which require to close 
+                // the polling request **after** the ping/pong handshake
+                if protocol == ProtocolVersion::V3 {
+                    socket.send(Packet::Noop)?;
+                }
+            }
+        }
+
```

Comment:
Also I think we don't need `cfg_if!` because we can do something like this almost everywhere where `cfg_if!` remains : 
```suggestion
  #[cfg(feature = "v3")]
  {
      // send a NOOP packet to any pending polling request so it closes gracefully
      // V3 protocol introduce _paused_ polling transport which require to close
      // the polling request **after** the ping/pong handshake
      if protocol == ProtocolVersion::V3 {
          socket.send(Packet::Noop)?;
      }
  }
```

So we can remove this dependency

Reply:
:open_mouth:  It didn't click in my head yet that this notation is possible (it makes sense though)! The `config_if` crate was recommended early on in the Rust docs so I just rolled with it. But your suggested approach gets the job done with less dependencies, which is indeed better. I'll change this as suggested.